src/include/access/tableam.h

```diff
@@ -1044,7 +1044,7 @@ table_scan_getnextslot(TableScanDesc sscan, ScanDirection direction, TupleTableS
 	 * CheckXidAlive for catalog or regular tables.  See detailed comments in
 	 * xact.c where these variables are declared.
 	 */
-	if (unlikely(TransactionIdIsValid(CheckXidAlive) && !bsysscan))
+	if (unlikely(TransactionIdIsValid(CheckXidAlive) && (sysscan_depth == 0)))
```

Comment:
open question, not a strong suggestion: would it make easier to follow if we had a macro for `sysscan_depth == 0`

like:
```C
#define SYSSCAN_IN_PROGRESS(sysscan_depth) (sysscan_depth > 0)

unlikely(TransactionIdIsValid(CheckXidAlive) && !SYSSCAN_IN_PROGRESS(sysscan_depth))
```

Reply:
Yes this looks much better, easier to understand.
I will change it this way.

Reply:
This is a good idea, I will add this Macro, thanks